includes/discovery/entity-physical/entity-physical.inc.php

```diff
@@ -190,6 +191,19 @@
         $entPhysicalIsFRU        = $entry['entPhysicalIsFRU'];
         $entPhysicalAlias        = $entry['entPhysicalAlias'];
         $entPhysicalAssetID      = $entry['entPhysicalAssetID'];
+
+        //VRP devices seems to use LogicalEntity '1' instead of '0' like the default code checks.
+        //Standard code is still run after anyway.
+        if (array_key_exists('1', $entry) && array_key_exists('entAliasMappingIdentifier', $entry['1'])) {
```

Comment:
You could use `isset($entry['1']['entAliasMappingIdentifier'])` here, it works on nested arrays.
Also, not sure about the duplication of the below code, it feels way to complex.

Maybe:
```
if (!empty($entry['1']['entAliasMappingIdentifier']) && str_contains('ifIndex',  $entry['1']['entAliasMappingIdentifier'])) {
    $ifIndex = str_replace('ifIndex.', '', $entry['1']['entAliasMappingIdentifier']);
}
```

not sure if it does the same thing, but yeah, below code is kinda silly.  I like the part where it uses strpos fIndex so it doesn't match position 0

Reply:
Ok. I'll clean it a little bit. 

Basically, a refactor would be needed. The entAliasMappingIdentifier is dual indexed, which gives this '1' or '0' or anything else really, according to the mib. 
What I would be interested to get is the lowest index of entAliasMappingIdentifier. 

Considering my dev plateform is down now (so I have to work on prod) I won't go for the refactor today.

Reply:
done with a regexp, and only for VRP for now.